client/taskenv/env.go

```diff
@@ -63,6 +69,9 @@ const (
 	// Region is the environment variable for passing the region in which the alloc is running.
 	Region = "NOMAD_REGION"
 
+	// DeploymentID is the environment variable for psasing the ID of the deployment that created the task
+	DeploymentID = "NOMAD_DEPLOYMENT_ID"
```

Comment:
Unfortunately the deployment ID is also not a fixed value. If a job is updated in place (ex. a canary meta change), then the deployment ID will be updated: https://github.com/hashicorp/nomad/blob/v0.12.5/scheduler/generic_sched.go#L386

Reply:
This has been a great learning experience about Nomad and I hope that I now understand the system (better). Thank you for the code pointer - this was very educating.